### PR TITLE
Adding grover featurizer

### PR DESCRIPTION
**Pull request by gusty1g**
In this PR, I am adding grover featurizer to suite of featurizers in deepchem.

### CODE REVIEW

File: deepchem/feat/molecule_featurizers/rdkit_descriptors.py [hunk 1]
```diff
@@ -27,24 +26,14 @@ class RDKitDescriptors(MolecularFeaturizer):
...
 
   Warning: Currently, the normalizing cdf parameters are not available for BCUT2D descriptors.
   (BCUT2D_MWHI, BCUT2D_MWLOW, BCUT2D_CHGHI, BCUT2D_CHGLO, BCUT2D_LOGPHI, BCUT2D_LOGPLOW, BCUT2D_MRHI, BCUT2D_MRLOW)
 
-  Attributes
-  ----------
-  descriptors: List[str]
-    List of RDKit descriptor names used in this class.
-
   Note
```
Comment (tonydavis629) [inline deepchem/feat/molecule_featurizers/rdkit_descriptors.py:37]: Maybe add some info about enabling custom descriptors.

**Review by tonydavis629 — CHANGES_REQUESTED** [review]
Looks good, I think we should figure out how to integrate the one hot encoder into the moleculer_featurizer one hot encoder. There's already 2 one hot encoders right now in deepchem. Also had a few questions that maybe a comment or little more detail would clarify.

File: deepchem/feat/molecule_featurizers/grover_featurizer.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+  Returns
+  -------
+  encoding: List[int]
+    A one-hot encoding of the value in a list of length len(choices) + 1.
+  If value is not in the list of choices, then the final element in the encoding is 1.
+  """
+  encoding = [0] * (len(choices) + 1)
+  index = choices.index(value) if value in choices else -1
+  encoding[index] = 1
+  return encoding
```
[the remark this answers is not in the document]
Comment (gusty1g) [reply deepchem/feat/molecule_featurizers/grover_featurizer.py:84]: I could use `dc.utils.molecule_feature_utils.one_hot_encode` which I did not know earlier. I will update PR to use it.

File: deepchem/feat/molecule_featurizers/grover_featurizer.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    'hybridization': [
+        Chem.rdchem.HybridizationType.SP,
+        Chem.rdchem.HybridizationType.SP2,
+        Chem.rdchem.HybridizationType.SP3,
+        Chem.rdchem.HybridizationType.SP3D,
+        Chem.rdchem.HybridizationType.SP3D2,
+    ]
+}
+
+_BOND_FDIM = 14
```
[the remark this answers is not in the document]
Comment (gusty1g) [reply deepchem/feat/molecule_featurizers/grover_featurizer.py]: In the latest change, I reused utilities from dmpnn featurizer. This removed the need for BOND_FDIM as well as ATOM_FDIM in grover featurizer. The grover featurizer is similar to DMPNNFeaturizer. In longer run, I think they both should be unified.

File: deepchem/feat/molecule_featurizers/grover_featurizer.py [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+class GroverFeaturizer(MolecularFeaturizer):
+  """Featurizer for GROVER Model
+
+  The Grover Featurizer is used to compute features suitable for grover model.
+  It accepts an rdkit molecule of type `rdkit.Chem.rdchem.Mol` or a SMILES string
+  as input and computes the following sets of features:
+    1. a molecular graph from the input molecule
+    2. functional groups which are used **only** during pretraining
+    3. additional features which can **only** be used during finetuning
```
[the remark this answers is not in the document]
Comment (gusty1g) [reply deepchem/feat/molecule_featurizers/grover_featurizer.py]: The pretraining task depends only on a specific set of features from the molecule - the function groups, the atom vocabulary and bond vocabulary. Hence, optional features are not required in pretraining. Note: Atom vocabulary and bond vocabulary are generated at the time of training and not included in this PR.

File: deepchem/feat/molecule_featurizers/rdkit_descriptors.py (hunk 1 above)
Comment (gusty1g) [reply deepchem/feat/molecule_featurizers/rdkit_descriptors.py]: I will make it in a different PR, improving overall docs.

**Comment by gusty1g** [comment]
A note to self: I need to fix mypy error and a failing unit test before merging.

**Comment by gusty1g** [comment]
> A note to self: I need to fix mypy error and a failing unit test before merging.

Fixed mypy and failing unit test. Going to merge this in @rbharath as this has been hanging around for some time now.

**Pull request merged**